### ISSUE TITLE
`Union` type alias not carrying more than one generic argument

### ISSUE THREAD

**Issue by betafcc** [microsoft/pyright#242]
When defining union of generic types, like so:

```python
from typing import TypeVar, Generic, Union, Callable
from dataclasses import dataclass

E = TypeVar("E")
A = TypeVar("A")
B = TypeVar("B")

@dataclass
class Left(Generic[E]):
    left: E

@dataclass
class Right(Generic[A]):
    right: A

Either = Union[Left[E], Right[A]]

def fmap(f: Callable[[A], B], either: Either[E, A]) -> Either[E, B]:
    if isinstance(either, Right):
        return Right(f(either.right))
    else:
        return either

def square(x: int) -> int:
    return x * x

reveal_type(fmap(square, Left("s")))
```

pyright complains when aplying the two generics argument to the `Either` union-alias:

![image](https://user-images.githubusercontent.com/27316303/64587621-52c8be00-d376-11e9-95a3-1e7ffb974ee2.png)

Yet, mypy seems  to recognize it:
![image](https://user-images.githubusercontent.com/27316303/64587665-75f36d80-d376-11e9-9eb9-d05a36515a28.png)

Not sure what the actual expected behavior should be in this case, because mypy worked I assumed that `Union` aliases should carry all the generics from its arguments.

Thanks for this awesome project btw

**Comment by erictraut** [comment]
Thanks for the bug report. This (a union of generic classes) strikes me as an obscure and odd case, but since mypy implements it, I'll do so as well in pyright. This will be implemented in the next release.

**Comment by erictraut** [comment]
This is implemented in 1.0.61, which I just published.

I also implemented support for the reveal_type call.
